client/src/actions/code.js

```diff
@@ -64,3 +67,37 @@ export function addCode(json) {
     }
 
 }
+
+export function selectCode(num, codesmell) {
+
+    return {
+        type: SELECT_CODE, 
+        payload: {
+            lineNumber: num, 
+            codeSmellName: codesmell
+        }
+    }
+}
+
+export function submitCode(userid, code, codesmells) {
+    return function (dispatch) {
+
+        return request
+            .post('http://localhost:8000/app/submit/', {
```

Comment:
Not really a fix request, but it doesn't need an authorization token? Disregard until we have everything working